src/utils/configOverride.ts

```diff
@@ -1,34 +1,84 @@
+
+export function getConfig() {
+  return formatSourceLocation(getConfigOverride() || defaultSource);
 }
 
-export function overrideConfig() {
-  // This code is only used for (integration) testing and will be optimized away in production builds
-  if (IS_DEV_BUILD) {
-    const url = new URL(window.location.href);
+function getConfigOverride() {
+  // Shortcut to determine if we can even use an override
+  if (allowedSources.length <= 0 && !unsafeAllowDynamicSources) {
+    return undefined;
+  }
+
+  const url = new URL(window.location.href);
+
+  if (url.searchParams.has(configFileQueryKey)) {
+    const configQuery = url.searchParams.get(configFileQueryKey)?.toLowerCase();
 
-    const configFile = url.searchParams.get(configFileQueryKey) || getStoredConfigOverride();
-    // Strip the config file query param from the URL since it's stored locally,
+    // Strip the config file query param from the URL and history since it's stored locally,
     // and then the url stays clean and the user will be less likely to play with the param
-    if (url.searchParams.has(configFileQueryKey)) {
-      url.searchParams.delete(configFileQueryKey);
+    url.searchParams.delete(configFileQueryKey);
+    window.history.replaceState(null, '', url.toString());
 
-      window.history.replaceState(null, '', url.toString());
+    // If the query param exists but the value is empty, clear the storage and allow fallback to the default config
+    if (!configQuery) {
+      clearStoredConfig();
+      return undefined;
     }
 
-    // Use session storage to cache any config location override set from the url parameter so it can be restored
-    // on subsequent navigation if the query string gets lost, but it doesn't persist if you close the tab
-    if (configFile) {
-      window.sessionStorage.setItem(configFileStorageKey, configFile);
+    // If it's valid, store it and return it
+    if (isValidConfigSource(configQuery)) {
+      storage.setItem(configFileStorageKey, configQuery);
+      return configQuery;
     }
 
-    window.configLocation = configFile ? `/test-data/config.${configFile}.json` : '/config.json';
+    // Yes this falls through to look up the stored value if the query string is invalid and that's OK
+  }
+
+  const storedSource = storage.getItem(configFileStorageKey)?.toLowerCase();
+
+  // Make sure the stored value is still valid before returning it
+  if (storedSource && isValidConfigSource(storedSource)) {
+    return storedSource;
   }
+
+  return undefined;
+}
+
+function isValidConfigSource(source: string) {
+  // Dynamic values are valid as long as they are defined
+  if (unsafeAllowDynamicSources) {
+    return !!source;
+  }
+
+  return allowedSources.indexOf(source) >= 0;
+}
+
+function formatSourceLocation(source?: string) {
+  if (!source) {
+    return undefined;
+  }
+
+  if (source.match(/^[a-z,\d]{8}$/)) {
+    return `https://cdn.jwplayer.com/apps/configs/${source}.json`;
```

Comment:
For dev environment the url will look like: 

```
https://content-portal.jwplatform.com/apps/configs/_ID_.json
```
Maybe we will be able just to change it in case we have APP_DEV_API env on.